src/components/EventUserTable.js

```diff
@@ -1,54 +1,300 @@
+        }
+      }
+      if (user.diet) {
+        dietary[user.diet] = dietary[user.diet] ? dietary[user.diet] + 1 : 1;
+      }
+      if (user.gender) {
+        genders[user.gender] = genders[user.gender] ? genders[user.gender] + 1 : 1;
+      }
+    })
+
+    this.setState({
+      faculties,
+      years,
+      genders,
+      dietary
+    });
+  }
+
+  async updateEventTableData(eventID) {
+    const eventParams = new URLSearchParams({
+      id: eventID
+    });
+
+    fetchBackend('/events/getUsers?' + eventParams, 'GET')
+      .then(response => response.json())
+      .then(async users => {
+        this.registrationNumbers(users)
       });
   }
 
   componentDidMount() {
     this.getEventTableData(this.props.event.id);
   }
 
+  /*
+    the if statement is only used if an exec goes from one event directly to another event (not implemented right now)
+  */
   componentDidUpdate(prevProps) {
     if (prevProps.event.id !== this.props.event.id) {
-      this.getEventTableData(this.props.event.id);
+      this.updateEventTableData(this.props.event.id);
+    }
+  }
+
+  /*
+    changes the visibility of the stats of whichever div was selected
+  */
+  changeVisibility(event) {
+    const id = event.target.id
+    const invisible = { visible: false, style: { display: 'none' } }
+    const visible = { visible: true, style: { display: 'flex', paddingBottom: '20px', paddingLeft: '50px' } }
+    switch (id) {
+      case 'registration':
+        if (this.state.registrationVisible.visible) {
+          this.setState({ registrationVisible: invisible })
+        } else {
+          this.setState({ registrationVisible: visible })
+        }
+        break;
+      case 'faculties':
+        if (this.state.facultyVisible.visible) {
+          this.setState({ facultyVisible: invisible })
+        } else {
+          this.setState({ facultyVisible: visible })
+        }
+        break;
+      case 'year':
+        if (this.state.yearVisible.visible) {
+          this.setState({ yearVisible: invisible })
+        } else {
+          this.setState({ yearVisible: visible })
+        }
+        break;
+      case 'dietary':
+        if (this.state.dietaryVisible.visible) {
+          this.setState({ dietaryVisible: invisible })
+        } else {
+          this.setState({ dietaryVisible: visible })
+        }
+        break;
+      case 'gender':
+        if (this.state.gendersVisible.visible) {
+          this.setState({ gendersVisible: invisible })
+        } else {
+          this.setState({ gendersVisible: visible })
+        }
+        break;
+      case 'heardFrom':
+        if (this.state.heardFromVisible.visible) {
+          this.setState({ heardFromVisible: invisible })
+        } else {
+          this.setState({ heardFromVisible: visible })
+        }
+        break;
+      default:
+        ;
     }
   }
 
   render() {
+    //each data set is an array of data (arrays) sets b/c different charts accept different data
```

Comment:
Is it possible or better to format the data when you get that data? (Inside registrationNumbers and notRegistrationNumbers)

Reply:
hmm what do you think about putting data in the state? and it gets updated everytime inside of registrationNumbers and notRegistrationNumbers

Reply:
Yup thats what I meant, do you think that'll be better/faster/cleaner?

Reply:
i think it'll be cleaner but im not sure about faster. it's recomputed/updated every single time the stats are updated either way so maybe it doesn't have a big difference speed wise